src/java.base/share/classes/java/util/GregorianCalendar.java

```diff
@@ -2972,6 +2978,56 @@ private boolean isCutoverYear(int normalizedYear) {
         return normalizedYear == cutoverYear;
     }
 
+    /**
+     * {@return {@code true} if the first week of the current year is minimum
+     * and the {@code DAY_OF_WEEK} does not exist in that week}
+     *
+     * This method is used to check the validity of a {@code WEEK_OF_YEAR} and
+     * {@code DAY_OF_WEEK} combo when WEEK_OF_YEAR is rolled to a value of 1.
+     * This prevents other methods from calling complete() with an invalid combo.
+     */
+    private boolean isInvalidWeek1() {
+        // Calculate the DAY_OF_WEEK for Jan 1 of the current YEAR
+        long jan1Fd =  gcal.getFixedDate(internalGet(YEAR), 1, 1, null);
+        int jan1Dow = BaseCalendar.getDayOfWeekFromFixedDate(jan1Fd);
+        int daysInFirstWeek;
+        if (getFirstDayOfWeek() <= jan1Dow) {
+            // Add wrap around days
+            daysInFirstWeek = (7 - jan1Dow) + getFirstDayOfWeek();
+        } else {
+            daysInFirstWeek = getFirstDayOfWeek() - jan1Dow;
+        }
+        // If the week is minimum, check if the DAY_OF_WEEK does not exist
+        return isMinWeek(daysInFirstWeek) &&
+                dayNotInMinWeek(internalGet(DAY_OF_WEEK), jan1Dow, getFirstDayOfWeek() - 1);
+    }
+
+    /**
+     * Determines if the specified amount of days can make up
+     * a valid minimum week.
+     */
+    private boolean isMinWeek (int days) {
+        return days >= getMinimalDaysInFirstWeek();
+    }
+
+    /**
+     * Determines if the specified day exists in the minimum week.
+     * For example, dayNotInMinWeek(4, 6, 3) returns false since Wednesday
+     * is not between the minimum week given by [Friday, Saturday,
+     * Sunday, Monday, Tuesday].
+     */
+    private boolean dayNotInMinWeek (int day, int startDay, int endDay) {
+        if (endDay >= startDay) {
+            // dayNotInMinWeek(2, 3, 6), check that 2 is
+            // not between 3 4 5 6
+            return !(day >= startDay && day <= endDay);
+        } else {
+            // dayNotInMinWeek(4, 6, 3), check that 4 is
+            // not between 6 7 1 2 3
+            return !(day >= startDay || day <= endDay);
+        }
+    }
+
```

Comment:
Are these `GregorianCalendar` specific? What about other calendars?

Reply:
Good point, these are _probably_ applicable to the other calendars, will take another look there.